src/include/rados/librados.hpp

```diff
@@ -1019,8 +1019,6 @@ namespace librados
     void set_assert_version(uint64_t ver);
     void set_assert_src_version(const std::string& o, uint64_t ver);
 
-    const std::string& get_pool_name() const;
-
```

Comment:
I think changing the signature for this function is problematic if we want to avoid breaking linkage.

We could keep an internal string list of all unique names we've seen for the pool and return const refs to those. pretty lame but ...